test/OpenTelemetry.Exporter.OneCollector.Tests/CommonSchemaJsonSerializationStateTests.cs

```diff
@@ -106,6 +139,48 @@ public void AddExtensionAttributeDuplicatesTest()
             json);
     }
 
+    [Fact]
+    public void AddExtensionAttributeMixedFieldAndPropertyTest()
+    {
+        using var stream = new MemoryStream();
+        using var writer = new Utf8JsonWriter(stream);
+
+        var state = new CommonSchemaJsonSerializationState("Test", writer);
+
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext. foo .foo_field1", 1));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo", new List<KeyValuePair<string, object?>> { new KeyValuePair<string, object?>(" foo.field2 ", 2) }));
```

Comment:
Should this line overwrite Ln150? If not, what's the rationale? (need to understand the design)

Reply:
The current design is extensions are building up a list of `KeyValuePair<string, object>`s.

Today you can do this:

```csharp
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field1", 1));
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field2", 2));
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field3", 3));
// output: "ext": { "foo": { "field1": 1, "field2": 2, "field3": 3}  }
```

After this PR you can do this:

```csharp
state.AddExtensionAttribute(new KeyValuePair<string, object?>(
   "ext.foo",
   new Dictionary<string, object>() { 
      ["field1"] = 1,
      ["field2"] = 2,
      ["field3"] = 3,
   }));
// output: "ext": { "foo": { "field1": 1, "field2": 2, "field3": 3 } }
```

If you mix and match these styles, the final list has everything in the order it came in:

```csharp
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field1", 1));
state.AddExtensionAttribute(new KeyValuePair<string, object?>(
   "ext.foo",
   new Dictionary<string, object>() { 
      ["field2"] = 2,
      ["field3"] = 3,
   }));
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field3", 300));
// output: "ext": { "foo": { "field1": 1, "field2": 2, "field3": 3, "field3": 300 } }
```

Note "field3" shows up twice in this case.

That is the same behavior as today if you did...

```csharp
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field3", 3));
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field3", 300));
// output: "ext": { "foo": { "field3":3, "field3": 300 } }
```

You would also get "field3" twice. There isn't de-duping of extensions because there isn't de-duping of attributes or scopes in logging either. It is just consistently bad across the board. For better or worse! De-duping could be added if users want to pay for the perf, but so far no one is complaining so I just kept it dumb/cheap.

If you try to push something which does not implement `IReadOnlyList` or `IEnumerable` of `KVP<string, object>`...

```csharp
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo", "some string");
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo", 0);
```

...it will be dropped with a log. We could support these (I did initially), but I decided to limit to ~maps because all the CS extensions currently defined are maps and it greatly reduces the complexity of the logic having the limitation.